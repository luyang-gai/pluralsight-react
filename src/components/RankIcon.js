import React, {PropTypes} from 'react';

const RankIcon = ({rank, tier}) => {
  return (
    <img src={`https://opgg-static.akamaized.net/images/medals/${tier}_${rank}.png`}/>
  );
};

RankIcon.propTypes = {
  rank: PropTypes.string.isRequired,
  tier: PropTypes.string.isRequired
};

export default RankIcon;
