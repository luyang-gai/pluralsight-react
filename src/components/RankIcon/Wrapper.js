import styled from 'styled-components';

const Wrapper = styled.img`
  height: 100%;
`;

export default Wrapper;
