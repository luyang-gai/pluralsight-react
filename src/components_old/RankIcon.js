import React, {PropTypes} from 'react';
import * as styles from './common/Styles';

const imageStyles = {
  height: "100%"
};

const containerStyles = {
  alignItems: "center",
  textAlign: "center",
  border: `2px solid ${styles.PRIMARY_COLOR}`,
  width: "200px",
  height: "120px",
  borderRadius: "4px",
  display: "flex"
};

const RankIcon = ({rank, tier, wins, losses, winRate}) => {
  return (
    <div style={containerStyles}>
      <img style={imageStyles} src={`https://opgg-static.akamaized.net/images/medals/${tier}_${rank}.png`}/>
      <div className="tier-rank-info">
        <div className="tier-rank">{tier} {rank}</div>
        <div className="">{wins} W {losses} L</div>
        <div className="win-rate">Win ratio {winRate}%</div>
      </div>
    </div>
  );
};

RankIcon.propTypes = {
  rank: PropTypes.string.isRequired,
  tier: PropTypes.string.isRequired
};

export default RankIcon;
