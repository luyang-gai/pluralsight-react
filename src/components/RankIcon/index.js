import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';

const RankIcon = (props) => {
  return (
    <Wrapper src={`https://opgg-static.akamaized.net/images/medals/${props.tier}_${props.rank}.png`}/>
  );
};

RankIcon.propTypes = {
  rank: PropTypes.string.isRequired,
  tier: PropTypes.string.isRequired
};

export default RankIcon;
