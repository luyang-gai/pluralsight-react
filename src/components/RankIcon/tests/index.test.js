import React from 'react';
import RankIcon from '../index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <RankIcon tier="silver" rank="5"/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
