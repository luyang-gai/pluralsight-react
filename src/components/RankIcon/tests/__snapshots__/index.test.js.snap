// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  className="ciYYTC"
  src="https://opgg-static.akamaized.net/images/medals/silver_5.png"
/>
`;
